pysumma/plotting/layers.py

```diff
@@ -0,0 +1,54 @@
+'''Layer plotting for SUMMA output'''
+
+import numpy as np
+import xarray as xr
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import itertools
+
+
+def layers(ds, var, cmap='viridis'):
+    '''Plot a cross section of layers'''
+    layers = ds.nLayers.values.astype('int')
+    max_layers = np.amax(layers)
+    depths = np.empty((max_layers+1, len(ds.time)))
+    depths[:] = np.nan
+    vals = np.empty((max_layers, len(ds.time)))
+    vals[:] = np.nan
+
+    # Look for the right dimension
+    layer_refs = ['Snow', 'Soil', 'Toto']
+    for ref in layer_refs:
+        test_coord = 'mid{}AndTime'.format(ref)
+        if test_coord in ds[var].dims:
+            ifcStartIdx = ds['ifc{}StartIndex'.format(ref)].values
+            midStartIdx = ds['mid{}StartIndex'.format(ref)].values
+            break
+    else:
+        raise ValueError("Dataset provided doesn't appear to have layers!")
+
+    # Unpack values for depth and desired variable
+    for i in range(len(ds['time'].values)):
+        start_ifc = int(ifcStartIdx[i]) - 1
```

Comment:
In this for statement, there was an error. I couldn't understand this for statement. ifcStartIdx[i] is a list of the number of layer. But i is the number of time value. So the length of each data is different.